web-ui/src/components/feedback_request_confirmation/FeedbackRequestConfirmation.jsx

```diff
@@ -1,28 +1,65 @@
-import React from "react";
+import React, { useContext } from "react";
 import Typography from "@material-ui/core/Typography";
 import CheckCircleIcon from '@material-ui/icons/CheckCircle';
-import {makeStyles} from '@material-ui/core/styles';
-
-
+import { makeStyles } from '@material-ui/core/styles';
+import { selectProfile } from "../../context/selectors";
+import { AppContext } from "../../context/AppContext";
+import { useLocation } from 'react-router-dom';
+import queryString from 'query-string';
+import DateFnsUtils from "@date-io/date-fns";
 import "./FeedbackRequestConfirmation.css";
-import {green} from "@material-ui/core/colors";
+import { green } from "@material-ui/core/colors";
+
+const dateUtils = new DateFnsUtils();
 
-const formatNamesAsList = (names) => {
-  return names.join(", ")
-}
 const useStyles = makeStyles({
-announcement: {
-  textAlign: "center"
-},
+  announcement: {
+    textAlign: "center"
+  },
 });
 
+let today = new Date();
+today = dateUtils.format(today, "yyyy-MM-dd");
+
 const FeedbackRequestConfirmation = (props) => {
   const classes = useStyles();
+  const { state } = useContext(AppContext);
+  const location = useLocation();
+  const query = queryString.parse(location?.search);
+  const forQuery = query.for?.toString();
+  const fromQuery = query.from?.toString();
+  const sendQuery = query.send?.toString();
+  const requestee = selectProfile(state, forQuery);
+  let recipientInfo = getRecipientNames();
```

Comment:
this doesn't complain about being called before initialization?

Reply:
Because it's defined with the function keyword it gets 'hoisted' so you can call it before its defined in the file without error.  Arrow functions aren't hoisted but functions defined with the function keyword are.  Basically JS is weird haha.